go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -419,4 +422,128 @@ var DoltMerge = []enginetest.ScriptTest{
+			"SELECT DOLT_CHECKOUT('-b', 'feature-branch')",
+			"INSERT INTO test VALUES (1, 1);",
+			"UPDATE test SET val=1000 WHERE pk=0;",
+			"SELECT DOLT_COMMIT('-a', '-m', 'this is a normal commit');",
+			"SELECT DOLT_CHECKOUT('main');",
+			"UPDATE test SET val=1001 WHERE pk=0;",
+			"SELECT DOLT_COMMIT('-a', '-m', 'update a value');",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT DOLT_MERGE('feature-branch')",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:    "SELECT count(*) from dolt_conflicts_test",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:    "SELECT DOLT_MERGE('--abort')",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:    "SELECT * FROM test",
+				Expected: []sql.Row{{0, 1001}},
+			},
+			{
+				Query:    "SELECT count(*) from dolt_conflicts_test",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:    "SELECT count(*) from dolt_status",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:    "SET dolt_allow_commit_conflicts = 0",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:          "SELECT DOLT_MERGE('feature-branch')",
+				ExpectedErrStr: doltdb.ErrUnresolvedConflicts.Error(),
+			},
+			{
+				Query:          "SELECT count(*) from dolt_conflicts_test", // Commit allows queries when flags are set.
+				ExpectedErrStr: doltdb.ErrUnresolvedConflicts.Error(),
+			},
+		},
+	},
+	{
+		Name: "DOLT_MERGE with conflicts can be aborted when autocommit is off",
+		SetUpScript: []string{
+			"CREATE TABLE test (pk int primary key, val int)",
+			"INSERT INTO test VALUES (0, 0)",
+			"SET autocommit = 0",
+			"SELECT DOLT_COMMIT('-a', '-m', 'Step 1');",
+			"SELECT DOLT_CHECKOUT('-b', 'feature-branch')",
+			"INSERT INTO test VALUES (1, 1);",
+			"UPDATE test SET val=1000 WHERE pk=0;",
+			"SELECT DOLT_COMMIT('-a', '-m', 'this is a normal commit');",
+			"SELECT DOLT_CHECKOUT('main');",
+			"UPDATE test SET val=1001 WHERE pk=0;",
+			"SELECT DOLT_COMMIT('-a', '-m', 'update a value');",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT DOLT_MERGE('feature-branch', '-m', 'this is a merge')",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:    "SELECT * from dolt_status",
+				Expected: []sql.Row{{"test", true, "modified"}, {"test", false, "conflict"}},
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_conflicts",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:    "SELECT DOLT_MERGE('--abort')",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:    "SELECT * from dolt_status",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_log",
+				Expected: []sql.Row{{3}},
+			},
+			{
+				Query:    "SELECT * FROM test ORDER BY pk",
+				Expected: []sql.Row{{0, 1001}},
+			},
+			{
+				Query:    "SELECT DOLT_CHECKOUT('-b', 'other-branch')",
+				Expected: []sql.Row{{0}},
+			},
+		},
+	},
+	{
+		Name: "DOLT_MERGE complains when a merge overrides local changes",
```

Comment:
Should have a test where you set the new session var to false and confirm you can't commit

Reply:
Exists at line 465. Can make the test name more clear